frontend/workflows/serverexperimentation/src/dialog.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import Button from "@material-ui/core/Button";
+import MuiDialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Typography from "@material-ui/core/Typography";
+
+interface ButtonProps {
+  label: string;
+  onAction: () => void;
+}
+
+interface DialogProps {
+  title: string;
+  content: string;
+  open: boolean;
+  onClose: () => void;
+  buttons: ButtonProps[];
+}
+
+const Dialog: React.FC<DialogProps> = ({ title, content, open, onClose, buttons }) => {
```

Comment:
@dschaller Per https://github.com/lyft/clutch/issues/415 I know that you want to introduce a dialog/modal component in Clutch at some point. Do you think that there would be any value in me moving the dialog defined in here to Clutch Core? I know that it does NOT look the way you would like it to look eventually but if we have an abstraction for dialogs in core you can change it later on (its internal implementation) to match your designs accordingly. We could keep the public API pretty limited (what I outlined above is almost a minimum I guess).

Let me know what you think.

Reply:
@Augustyniak I think this is a great candidate for core. We can take in the relevant props and then render children within the actions